holoviews/core/dimension.py

```diff
@@ -158,6 +158,13 @@ def value_format(self, specs=None, **values):
     def range(self, specs=None, **values):
         return self._redim('range', specs, **values)
 
+    def label(self, specs=None, **values):
+        for k, v in values.items():
+            dim = self.parent.get_dimension(k)
```

Comment:
``redim`` doesn't validate Dimensions so you can apply it nested datastructures. That said I'd possibly be open to tightening that up when explicit ``specs`` are supplied.